### PR TITLE
CI: run the metal CI nightly & on PRs touching the config file

### PR DESCRIPTION
**Pull request by jakevdp**
Had to remove one test because the behavior of `jnp.clip` changed recently.

### CODE REVIEW

File: .github/workflows/metal_plugin_ci.yml
Comment (skye) [inline .github/workflows/metal_plugin_ci.yml:8]: Are we ready to use self-hosted runners on PRs? This workflow runs on a self-hosted mac runner, so running on PRs may allow someone to run malicious or spammy code on the runner. This is alleviated by us having to approve workflow runs by outside contributors: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

I'm personally ok with this, but wanna make sure @shuhand0 also is (who owns the runner). The advantage of this is I think we can add Shuhan and others as low-permission members of the jax repo, and then they can use PRs to test things out on the runner.

Jake, this change also LGTM if you remove this pull_request part, if you wanna go ahead and submit now and deal with PR runs later.

Comment (jakevdp) [reply .github/workflows/metal_plugin_ci.yml:8]: I think this is safe, because (1) this only triggers on pull requests that touch this yaml file, and (2) our CI is configured so that PRs from non-owners require owner approval before any actions execute.

Comment (shuhand0) [reply .github/workflows/metal_plugin_ci.yml:8]: Looks good to me.

**Review by shuhand0 — APPROVED** [review]
LGTM.

**Pull request merged**
